Frontpage

amb8805 opened the pull request:
Added functionality of the search form. Some placeholders, disabling multiple copies.  adding new fields.  Some basic form validation.

lukecoy committed:
Remove user route, add data_cite to repo

lukecoy committed:
Merge branch 'architecture' into oai_harvesting

lukecoy committed:
Merge architecture branch into harvesting branch

lukecoy committed:
functionality for the http client to datacite and the mapper

amb8805 committed:
adding new search bars and functioning dropdown fields while reworking event listener logic to use delegation

amb8805 committed:
removing unnecessary and unused case

amb8805 committed:
removing unnecessary lines, adding standardized reusable search bar to clone

varzaman committed:
Added a comment, just a test to make sure pushing works.

lukecoy committed:
Merge master in

lukecoy committed:
Merge branch 'oai_harvesting' of https://github.com/mozillascience/software-discovery-dashboard into oai_harvesting

amb8805 committed:
front-end dropdown and add logic and functionality

amb8805 committed:
merge master into frontpage

amb8805 committed:
fixed array logic

amb8805 committed:
removing ES7

varzaman commented on views/layout.jade:
```diff
@@ -20,17 +20,96 @@ html
+                    default:
+                      el.type="text";
+                      el.placeholder="";
+                      //el.required="required";
```
Any particular reason for this commented out line?

kocsenc commented:
There seems to be a lot of logic and in `layout.jade` - could be a problem in the future.

amb8805 committed:
fixes pull request concerns

amb8805 commented:
Most issues should be repaired in r2efd368

amb8805 committed:
moving scripts from layout.jade to hello.js

amb8805 committed:
fixing disabled dropdown functionality

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+      searchBarNodeClone.childNodes[0].childNodes[0].innerHTML = "Select " + "<span class=\"caret\"></span>";
+      searchBarNodeClone.childNodes[1].value="";
+      searchBarNodeClone.childNodes[1].placeholder="";
+      event.target.parentNode.parentNode.parentNode.insertBefore(searchBarNodeClone,event.target.parentNode.parentNode.nextSibling);
```
Can't you just do a document.getElement call to get the node?

amb8805 replied:
No since these listeners are added dynamically and the context where they are added is dependent on the element an ID would be no benefit, as it would need to be added dynamically then stored somewhere for reference, which provides more overhead than actually necessary

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+      searchBarNodeClone.childNodes[0].childNodes[0].innerHTML = "Select " + "<span class=\"caret\"></span>";
+      searchBarNodeClone.childNodes[1].value="";
+      searchBarNodeClone.childNodes[1].placeholder="";
+      event.target.parentNode.parentNode.parentNode.parentNode.insertBefore(searchBarNodeClone,event.target.parentNode.parentNode.parentNode.nextSibling);
```
Same as my other comment about document.getElement.

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+      event.target.parentNode.parentNode.parentNode.parentNode.insertBefore(searchBarNodeClone,event.target.parentNode.parentNode.parentNode.nextSibling);
+    }
+  //dropdown
+    else if(event.target.nodeName == 'A'){
```
nodeName should already be a string, so you don't need type conversion(nor should you want it in this case). Use === instead.

amb8805 replied:
fixed

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+  //dropdown
+    else if(event.target.nodeName == 'A'){
+    	if(!event.target.parentNode.classList.contains('disabled')){
+    		var keywordButton = event.target.parentNode.parentNode.parentNode.firstChild;
```
spacing before var

amb8805 replied:
should be fixed

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+	              el.placeholder="";
+	          }
+	        }
+    		});
```
Parentheses line 71 and 72 shouldn't be lined up.

amb8805 replied:
should've been fixed in recent commit

amb8805 committed:
fixes for pull request

amb8805 committed:
spacing

lukecoy commented:
for all the javascript folks - consider `let` and `const` instead of `var` in the future

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const

http://stackoverflow.com/questions/762011/let-keyword-vs-var-keyword

@Lettuceman44 @mok4ry

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+	          }
+	        }
+	      });
+				disabledKeywordHelper();
```
formatting issues

amb8805 committed:
sublime spacing

varzaman commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+	        }
+	      });
+	      disabledKeywordHelper();
+			}
```
parentheses formatting

amb8805 committed:
Manual spacing fix

nokeeo commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+  //add button icon
+    else if(event.target.parentNode.classList.contains('add-search-button')){
+      var searchBarNodeClone = document.querySelectorAll('div.reusable-search')[0].cloneNode(true);
+      searchBarNodeClone.childNodes[0].childNodes[0].innerHTML = "Select " + "<span class=\"caret\"></span>";
```
Seems like this code is duplicated from the add button.  Does not conform to the principles of DRY. This may cause issues in the future.

nokeeo commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+  //add button
+    if(event.target.classList.contains('add-search-button')){
+      var searchBarNodeClone = document.querySelectorAll('div.reusable-search')[0].cloneNode(true);
+      searchBarNodeClone.childNodes[0].childNodes[0].innerHTML = "Select " + "<span class=\"caret\"></span>";
```
If searchBarNodeClone does not have any children this code will crash which could be catastrophic.

nokeeo commented on public/javascripts/hello.js:
```diff
@@ -1,3 +1,97 @@
+  //dropdown
+    else if(event.target.nodeName === 'A'){
+      if(!event.target.parentNode.classList.contains('disabled')){
+    	var keywordButton = event.target.parentNode.parentNode.parentNode.firstChild;
```
Could we use ES11's .greatUncle accessor here instead of .parentNode.parentNode.parentNode.firstChild
Much cleaner.

lukecoy committed:
Figshare interface for searching

Figshare client and mapper for querying figshare with jsonld
Also adds auth package for local storing of auth credentials
when interfacing with research repos server side

lukecoy committed:
remove search validation for now

lukecoy committed:
spacing instead of tabs

amb8805 committed:
Styling footer and adding form submission.

amb8805 committed:
isempty

Commit:
add first version of results page

lukecoy committed:
Convert returned figshare results to jsonLD w/ context

Commit:
Merge branch 'oai_harvesting' into frontpage
